diff --git a/meson.build b/meson.build
index d4c75a907f..3512001dc9 100644
--- a/meson.build
+++ b/meson.build
@@ -26,8 +26,10 @@ libavutil = dependency('libavutil', version: '>= 58.29.100')
 libswresample = dependency('libswresample', version: '>= 4.12.100')
 libswscale = dependency('libswscale', version: '>= 7.5.100')
 
+libplacebo_opt = get_option('libplacebo')
 libplacebo = dependency('libplacebo', version: '>=6.338.2',
-                default_options: ['default_library=static', 'demos=false'])
+                default_options: ['default_library=static', 'demos=false'],
+                required: libplacebo_opt)
 
 libass = dependency('libass', version: '>= 0.12.2')
 
@@ -38,17 +40,20 @@ dependencies = [libass,
                 libavfilter,
                 libavformat,
                 libavutil,
-                libplacebo,
                 libswresample,
                 libswscale]
 
+if libplacebo.found()
+    dependencies += libplacebo
+endif
+
 # Keeps track of all enabled/disabled features
 features = {
     'debug': get_option('debug'),
     'ffmpeg': true,
     'gpl': get_option('gpl'),
     'libass': true,
-    'libplacebo': true,
+    'libplacebo': libplacebo.found(),
 }
 
 
@@ -250,12 +255,6 @@ sources = files(
     'video/repack.c',
     'video/sws_utils.c',
 
-    ## libplacebo
-    'video/out/placebo/ra_pl.c',
-    'video/out/placebo/utils.c',
-    'video/out/vo_gpu_next.c',
-    'video/out/gpu_next/context.c',
-
     ## osdep
     'osdep/io.c',
     'osdep/subprocess.c',
@@ -267,6 +266,13 @@ sources = files(
     'ta/ta_utils.c',
 )
 
+if libplacebo.found()
+    sources += files('video/out/placebo/ra_pl.c',
+                     'video/out/placebo/utils.c',
+                     'video/out/vo_gpu_next.c',
+                     'video/out/gpu_next/context.c')
+endif
+
 
 # compiler stuff
 cc = meson.get_compiler('c')
@@ -1303,10 +1309,13 @@ endif
 
 
 # vulkan
-vulkan_opt = get_option('vulkan').require(
-    libplacebo.get_variable('pl_has_vulkan', default_value: '0') == '1',
-    error_message: 'libplacebo compiled without vulkan support!',
-)
+vulkan_opt = get_option('vulkan')
+if features['libplacebo'] and vulkan_opt.enabled()
+    vulkan_opt = vulkan_opt.require(
+        libplacebo.get_variable('pl_has_vulkan', default_value: '0') == '1',
+        error_message: 'libplacebo compiled without vulkan support!',
+    )
+endif
 # libplacebo/ffmpeg public headers depend on Vulkan 1.3
 vulkan = dependency('vulkan', version: '>= 1.3.238', required: vulkan_opt)
 features += {'vulkan': vulkan.found() and (vulkan.type_name() == 'internal' or
diff --git a/meson.options b/meson.options
index 836d16d03f..db71e18108 100644
--- a/meson.options
+++ b/meson.options
@@ -17,6 +17,7 @@ option('javascript', type: 'feature', value: 'auto', description: 'Javascript (M
 option('jpeg', type: 'feature', value: 'auto', description: 'libjpeg image writer')
 option('lcms2', type: 'feature', value: 'auto', description: 'LCMS2 support')
 option('libarchive', type: 'feature', value: 'auto', description: 'libarchive wrapper for reading zip files and more')
+option('libplacebo', type: 'feature', value: 'auto', description: 'libplacebo GPU renderer')
 option('libavdevice', type: 'feature', value: 'auto', description: 'libavdevice')
 option('libbluray', type: 'feature', value: 'auto', description: 'Bluray support')
 option('lua',
